Source/DataModels/Components/ComponentMeshCollider.cpp

```diff
@@ -0,0 +1,194 @@
+#include "ComponentMeshCollider.h"
+
+#include "GameObject/GameObject.h"
+#include "Application.h"
+
+#include "FileSystem/Json.h"
+#include "ComponentTransform.h"
+
+#include "Geometry/LineSegment.h"
+#include "Geometry/Ray.h"
+#include "Physics/Physics.h"
+
+ComponentMeshCollider::ComponentMeshCollider(bool active, GameObject* owner)
+	: Component(ComponentType::MESHCOLLIDER, active, owner, true),
+	isTrigger(false)
+{
+}
+
+ComponentMeshCollider::~ComponentMeshCollider()
+{
+}
+
+void ComponentMeshCollider::Update()
+{
+}
+
+void ComponentMeshCollider::Draw()
+{
+
+}
+
+void ComponentMeshCollider::SaveOptions(Json& meta)
+{
+	// Do not delete these
+	meta["type"] = GetNameByType(type).c_str();
+	meta["active"] = (bool)active;
+	meta["removed"] = (bool)canBeRemoved;
+}
+
+void ComponentMeshCollider::LoadOptions(Json& meta)
+{
+	// Do not delete these
+	type = GetTypeByName(meta["type"]);
+	active = (bool)meta["active"];
+	canBeRemoved = (bool)meta["removed"];
+}
+
+bool ComponentMeshCollider::Move(Direction direction, float size, RaycastHit& hit, float stepSize )
+{
+	ComponentTransform* trans = static_cast<ComponentTransform*>(owner->GetComponent(ComponentType::TRANSFORM));
+	float3 position = trans->GetPosition();
+
+	math::vec points[8];
+	trans->GetObjectOBB().GetCornerPoints(points);
+	std::vector<float3> PositionPoints = { points[0], points[1],  points[2],  points[3],  points[4],  points[5],  points[6],  points[7]};

```

Comment:
Just a suggestion, what you did is perfectly fine.
```suggestion
	std::vector<float3> PositionPoints(std::begin(points), std::end(points));
```